level1/f256/modules/vtio.asm

```diff
@@ -1416,6 +1428,333 @@ SSRelea             lda       PD.CPR,y            get the current process ID
+                    sta       MAPSLOT             store it in the MMU slot to map it in
+* 		    **** Calculate starting address at 1000,1008,1010
+		    ldb	      R$Y+1,x               ldb with bitmap#
+		    lda	      #$08		  multiply by 8 (to start at 0, 8 or 16)
+		    mul	      			  d should be 0,8,or 16
+		    addd      #MAPADDR
+		    addd      #$1000
+		    tfr	      d,y		  y is address of BM(0-2) registers
+		    ldd	      R$X,x		  d now has CLUT#
+		    orcc      #1		  Set carry bit
+		    rolb			  shift B, and rotate in enable it
+		    andcc     #$FE		  Clear carry bit
+* 		    **** Load Bitmap start block physical address into Vicky BM0, BM1 or BM2
+		    stb	      ,y           	  enable bitmap with CLUT R$X
+		    puls      a
+		    sta	      MAPSLOT
+		    puls      cc
+		    rts
+
+
+
+;;; SS.DfPal
+;;; Define Palette and populate a CLUT from Memory Module
+;;;
+;;; Entry: R$X = CLUT # (0-3)
+;;; 	   R$Y = pointer to location of data in caller process
+;;;        (Caller must load data module)
+;;;
+;;; Exit:  B = A non-zero error code.
+;;;       CC = Carry flag clear to indicate success
+SSDfPal	       	    pshs      a,x,y,u
+*		    **** Map in $C1 for CLUT Registers
+		    pshs      cc
+		    orcc      #IntMasks
+		    ldy	      <D.Proc
+		    pshs      y,x
+		    ldy	      <D.SysPrc
+		    sty	      <D.Proc
+		    ldb	      #$01
+		    ldx	      #$C1
+		    os9	      F$MapBlk
+		    bcs       errnomap@
+		    puls      y,x
+		    sty	      <D.Proc
+		    puls      cc
+*		    **** Calculate CLUT offset		    
+		    pshs      u	                  push map logical addr
+         	    lda       R$X+1,x
+                    lsla                          multiply by 2 so index works
+                    pshs      x			  push pointer to caller Regs
+                    leax      clutlookup,pcr
+		    ldd	      a,x
+                    leau      d,u                 ldx with $1000 offset for CLUT0
+*		    **** Start F$Move (with U from above)
+		    puls      x			  pull pointer to caller Regs
+		    pshs      x
+		    ldx       R$Y,x               x=Get pointer to caller data
+		    ldy	      <D.Proc		  Get caller process
+		    lda	      P$Task,y		  a=source Task# (Caller)
+		    ldb	      <D.SysTsk	      	  b=dest Task# (System)
+                    ldy	      #$400	          moving 1K
+                    os9       F$Move              copy data
+		    bcs       errormove@          return if error
+*		    **** Exit Move
+		    puls      x
+noerror@	    puls      u
+		    bsr	      clearblock
+		    andcc     #$FE
+		    clrb			  No Error Code
+		    bra	      end@
+errnomap@	    puls      y,x		  Come here on F$MapBlk error
+		    sty	      <D.Proc
+		    puls      cc
+		    orcc      #Carry	          Set carry bit on erro
+		    bra	      end@
+errormove@	    puls      x
+		    puls      u			  come here on F$Move erre
+		    bsr	      clearblock
+		    orcc      #Carry	          Set carry bit on erro
```

Comment:
Change to `coma` to clear the carry, thus saving 1 byte (and spell error correctly)